src/base/Ident.zig

```diff
@@ -52,43 +52,34 @@ pub const Attributes = packed struct(u3) {
 
 /// An interner for identifier names.
 pub const Store = struct {
-    interner: SmallStringInterner,
+    interner: SmallStringInterner = .{},
     /// The index of the local module import that this ident comes from.
     ///
     /// By default, this is set to index 0, the primary module being compiled.
     /// This needs to be set when the ident is first seen during canonicalization
     /// before doing anything else.
-    exposing_modules: std.ArrayList(ModuleImport.Idx),
-    attributes: std.ArrayList(Attributes),
-    next_unique_name: u32,
-
-    pub fn init(gpa: std.mem.Allocator) Store {
-        return Store{
-            .interner = SmallStringInterner.init(gpa),
-            .exposing_modules = std.ArrayList(ModuleImport.Idx).init(gpa),
-            .attributes = std.ArrayList(Attributes).init(gpa),
-            .next_unique_name = 0,
-        };
-    }
-
-    pub fn deinit(self: *Store) void {
-        self.interner.deinit();
-        self.exposing_modules.deinit();
-        self.attributes.deinit();
+    exposing_modules: std.ArrayListUnmanaged(ModuleImport.Idx) = .{},
+    attributes: std.ArrayListUnmanaged(Attributes) = .{},
+    next_unique_name: u32 = 0,
+
+    pub fn deinit(self: *Store, gpa: std.mem.Allocator) void {
+        self.interner.deinit(gpa);
+        self.exposing_modules.deinit(gpa);
+        self.attributes.deinit(gpa);
     }
 
-    pub fn insert(self: *Store, ident: Ident, region: Region) Idx {
-        const idx = self.interner.insert(ident.raw_text, region);
-        self.exposing_modules.append(@enumFromInt(0)) catch exitOnOom();
-        self.attributes.append(ident.attributes) catch exitOnOom();
+    pub fn insert(self: *Store, gpa: std.mem.Allocator, ident: Ident, region: Region) Idx {
+        const idx = self.interner.insert(gpa, ident.raw_text, region);
+        self.exposing_modules.append(gpa, @enumFromInt(0)) catch |err| exitOnOom(err);
```

Comment:
Does zig have a shorthand for this? I assume not, but would be nice if it could just be `catch existOnOom`

Reply:
I don't believe so, since `catch exitOnOom` would return a function pointer from the expression. Unfortunate.